persistence-cassandra/javadsl/src/test/scala/com/lightbend/lagom/javadsl/persistence/cassandra/CassandraPersistenceSpec.scala

```diff
@@ -34,13 +37,19 @@ class CassandraPersistenceSpec(system: ActorSystem) extends ActorSystemSpec(syst
   override def beforeAll(): Unit = {
     super.beforeAll()
 
+    // Join ourselves - needed because the Cassandra offset store uses cluster startup task
+    val cluster = Cluster(system)
+    cluster.join(cluster.selfAddress)
+
+    // first ensure that this node is Up (ie: remote port is properly bound)
```

Comment:
remote port is bound when the ActorSystem is started (before `apply` returns)

Reply:
You are completely right about this. Somehow I convinced myself that the port would only find when trying to form the cluster. This doesn't make any sense.

Reply:
That's getting really puzzling. 

The erroing test logs says:

```scala
 Remoting started with transport [Artery tcp]; listening on address [akka://PersistentEntityTestDriverCompatSpec@127.0.0.1:34359] with UID [3966907252480615512]
[info] PersistentEntityTestDriverCompatSpec: Starting Cassandra on port client port: 34359
```

So, artery bound on TCP port 34359 at actor system bootstrap. Fine. 
Then Cassandra launcher tries to pick a free port and OS returns 34359, which then conflicts when we try to start the Cassandra client. 
That doesn't make sense to me. Unless the actor system is not yet bootstrapped when we start the CassandraLauncher. 

I can only explain it with the following order of events: 

1. ActorSystem reserves port 34359 and releases the socket
1. CassandraLauncher reserves port 34359 and releases the socket
1. Artery starts on port 34359
1. CassandraLauncher starts and fails to bind on port 34359

(1 and 2 can be swapped)

Maybe forming the cluster before CassandraLauncher is exaggerated, but we should not try to reserve a port for Cassandra before we can confirm that Artery is running.

Reply:
If that scenario is true there is something wrong with the initialization in Artery. It's supposed to bind (and block until completed) before ActorSystem.apply returns.

Reply:
Oh, that's useful information. Although it makes it scarier now.

Reply:
I found it!

In the init of the test, we reserve the Cassandra port using `CassandraLauncher.randomPort` (opens socket, pick port, close socket). 

Then the ActorSystem bootstraps and take the same port. Artery bounds to it.

Then CassandraLauncher has the port already fixed and tries to bound to it.

Reply:
We can see it here:
https://github.com/lagom/lagom/pull/2804/files#diff-06bb277ec92c7db81e68c50ae7478f00R28

Reply:
ah, nice catch